src/Common/Allocator.cpp

```diff
@@ -44,9 +46,20 @@ auto adjustToPageSize(void * buf, size_t len, size_t page_size)
 }
 #endif
 
+bool madviseSupportsMadvPopulateWrite()
+{
+    VersionNumber supported(5, 14, 0);
```

Comment:
Let's also add a comment that it is not possible to rely on `madvise(MADV_POPULATE_WRITE) == EINVAL` for detection, since it will be returned in lots of other errros